include/libdnf-cli/output/transaction_table.hpp

```diff
@@ -125,11 +126,8 @@ class ActionHeaderPrinter {
 private:
     struct libscols_table * table = nullptr;
     struct libscols_line * current_header_line = nullptr;
-    // TODO(lukash) better default value?
-    libdnf::transaction::TransactionItemAction current_action =
-        libdnf::transaction::TransactionItemAction::REASON_CHANGE;
-    libdnf::transaction::TransactionItemReason current_reason =
-        libdnf::transaction::TransactionItemReason::DEPENDENCY;  // TODO(lukash) fix default values
+    std::optional<libdnf::transaction::TransactionItemAction> current_action;
+    std::optional<libdnf::transaction::TransactionItemReason> current_reason;
```

Comment:
Again, what are you asking about? Are you asking why I've used `std::optional` here? I hope it's quite obvious from the diff. The values are used in a loop to check whether the action/reason has changed from the previous item. The variables had an arbitrary default value that could just be equal to the action/reason of the first item in the looped-over container and the mechanism would fail. Since the enums don't allow any other value, I've used `std::optional` to differentiate that state. It's not the prettiest but it does the job fine and it's explicit about it.